### PR TITLE
Tests: Improve code coverage for `_build_block_template_result_from_file`

### PR DESCRIPTION
**Pull request by gziolo**
Trac ticket: https://core.trac.wordpress.org/ticket/59325

Follow-up to the comment from @costdev in https://github.com/WordPress/wordpress-develop/pull/5186#discussion_r1324147112:

> Here's a coverage report which includes existing tests and tests included in this PR.
> 
> image
> image
> 
> Everything that this PR intends to cover appears to be covered. 🙂
> 
> In terms of the remaining coverage for this function:
> - Line 585 isn't hit by the tests that specifically cover this function.
> - Two branches aren't covered.
>   - One branch can be covered by testing with a `postTypes` key in `$template_file` and a `$template_type` of `wp_template`.
>   - One branch can be covered by ensuring both outcomes of line 572 are met by tests, then testing `$template->title` for the expected result.
> - There don't appear to be tests specifically for this function that provide a non-default slug in `$template_file`. This means lines 578-581, while _hit_ by the tests, don't appear to have coverage to ensure that `$template->title` is retained for `wp_template` types with a non-default slug in the `$template_file`.
> 
> As the uncovered parts don't seem to be part of this PR's goals, we can follow up to complete coverage for `_build_block_template_result_from_file()`. I'd also suggest that we start to separate the tests for each function into their own files, completing `@ticket`, `@covers`, and any other annotations that need to be added. This makes it easier not to get lost amongst the tests.

All cases listed should be covered with this PR.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/block-template-utils.php [hunk 1]
```diff
@@ -102,8 +107,14 @@ public function test_build_block_template_result_from_post() {
 		$this->assertSame( 'Description of my template', $template->description );
 		$this->assertSame( 'wp_template', $template->type );
 		$this->assertSame( self::$template_post->post_modified, $template->modified, 'Template result properties match' );
+	}
 
-		// Test template parts.
+	/**
+	 * @ticket 54335
```
Comment (ockham) [inline tests/phpunit/tests/block-template-utils.php:113]:
```suggestion
	 * @ticket 54325
```

Comment (ockham) [reply tests/phpunit/tests/block-template-utils.php:113]: 🔢 Some more instances of this below.

File: tests/phpunit/tests/block-template-utils.php [hunk 2]
```diff
@@ -86,7 +86,12 @@ public function set_up() {
 		switch_theme( self::TEST_THEME );
 	}
 
-	public function test_build_block_template_result_from_post() {
+	/**
+	 * @ticket 54335
+	 *
+	 * @covers ::_build_block_template_result_from_post
+	 */
+	public function test_build_block_template_result_from_post_template() {
```
Comment (ockham) [inline tests/phpunit/tests/block-template-utils.php]: The name's a bit hard to read; how about we keep this as it was before

```suggestion
	public function test_build_block_template_result_from_post() {
```

and change the new one from `test_build_block_template_result_from_post_template_part` to `test_build_block_template_part_result_from_post_template`?

I realize that would mean that it's no longer following the `test_{function_name}_xyz` pattern, but maybe that's okay, given that we have the `@covers` PHPDoc to indicate the function it's covering.

File: tests/phpunit/tests/block-template-utils.php [hunk 3]
```diff
@@ -102,8 +107,14 @@ public function test_build_block_template_result_from_post() {
...
 		$this->assertSame( self::$template_post->post_modified, $template->modified, 'Template result properties match' );
+	}
 
-		// Test template parts.
+	/**
+	 * @ticket 54335
+	 *
+	 * @covers ::_build_block_template_result_from_post
+	 */
+	public function test_build_block_template_result_from_post_template_part() {
```
Comment (ockham) [inline tests/phpunit/tests/block-template-utils.php]: See https://github.com/WordPress/wordpress-develop/pull/5396/files#r1345811456.

🔢 Similar instances below (`test_build_block_template_result_from_file_template` etc)

**Review by ockham — APPROVED** [review]
Left a minor note about test method names, but nothing blocking.

Thank you! LGTM 😊

File: tests/phpunit/tests/block-template-utils.php (hunk 2 above)
Comment (costdev) [reply tests/phpunit/tests/block-template-utils.php]: What about moving the respective tests to their own files and using the `test_should_` / `test_should_not_` format ([Ref](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#test-methods)) for test names, which also reduces redundant references to the function's name?

i.e.

Filename: `buildBlockTemplateResultFromPost.php`
Classname: `Tests_Block_Template_Utils_BuildBlockTemplateResultFromPost`
Tests:
  - `test_should_build_template()`
  - `test_should_build_template_part()`
---
Filename: `buildBlockTemplateResultFromFile.php`
Classname: `Tests_Block_Template_Utils_BuildBlockTemplateResultFromFile`
Tests:
  - `test_should_build_template()`
  - `test_should_build_template_using_custom_properties()`
  - `test_should_enforce_default_properties_when_building_template()`
  - `test_should_respect_post_types_property_when_building_template()`
  - `test_should_build_template_part()`
  - `test_should_ignore_post_types_property_when_building_template_part()`

Using this approach may also help with identifying possible additional tests.

**Comment by gziolo** [comment]
Thank you for the feedback. I’ll rename and move test cases as suggested and land tomorrow 👍

File: tests/phpunit/tests/block-template-utils.php (hunk 2 above)
Comment (gziolo) [reply tests/phpunit/tests/block-template-utils.php]: I refactored the code as recommended with f0250e35660ab87bd2c78c497908365a8d9d278d. Let me know if it that works for you.

File: tests/phpunit/tests/block-template-utils.php (hunk 3 above)
Comment (gziolo) [reply tests/phpunit/tests/block-template-utils.php]: It should be better after f0250e35660ab87bd2c78c497908365a8d9d278d.

**Review by costdev — APPROVED** [review]
Thanks for the updates @gziolo! This looks good to me!

Coverage shows the `hooked_blocks()` lines aren't covered by these tests.

It would be good to cover these two conditions so there's a complete set of tests in this function's test class. Even if covered by tests for another function, having tests here means that if these tests pass and the other function's fail, we know to look in the other function right away.

image

That said, the final coverage doesn't need to be done in this PR and could be a follow-up if you'd like.

Approving 🙂

**Comment by gziolo** [comment]
> It would be good to cover these two conditions so there's a complete set of tests in this function's test class.

Do I understand correctly that you mean adding coverage in the same test suite? I'm sure it has coverage in other places, but I agree we can add another test here 👍🏻

I want to wait until #5455 as I will have to rebase this branch and probably remove one or two tests, maybe move some new tests. I can work on the feedback as part of that.

**Comment by gziolo** [comment]
Committed with https://core.trac.wordpress.org/changeset/56983.

> It would be good to cover these two conditions so there's a complete set of tests in this function's test class.

I will work on more tests separately covering the filter `hooked_block_types`, and I will tackle it as part of that effort.

**Pull request closed without merging**